dlt/extract/utils.py

```diff
@@ -60,14 +62,21 @@
     pandas = None
 
 
-def get_data_item_format(items: TDataItems) -> TDataItemFormat:
+def get_data_item_format(items: TDataItems, meta: Any = None) -> TDataItemFormat:
```

Comment:
Each row that has a special item type needs to have the meta set, similar to how the ImportFileMeta works. We could also store it so it works more like the other hints, but I think this should be ok. It's mostly going to be used from the transformations anyway.

Reply:
as mentioned, just check the item type